v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compiler/index.ts

```diff
@@ -77,17 +77,22 @@ export async function downloadConfiguredCompilers(
   }
 }
 
-export async function getCompiler(version: string): Promise<Compiler> {
-  const platform = CompilerDownloaderImplementation.getCompilerPlatform();
-  const compilerDownloader = new CompilerDownloaderImplementation(
-    platform,
-    await getGlobalCompilersCacheDir(),
-  );
+export async function getCompiler(
+  version: string,
+  preferWasm: boolean,
+): Promise<Compiler> {
+  if (preferWasm === false) {
```

Comment:
`preferWasm` is not nullable nor optional, so `!preferWasm` should be fine here.